entity-management-definitions/src/main/java/eu/europeana/entitymanagement/normalization/EntityFieldsMinimalValidator.java

```diff
@@ -0,0 +1,59 @@
+package eu.europeana.entitymanagement.normalization;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.springframework.stereotype.Component;
+
+import eu.europeana.entitymanagement.definitions.exceptions.EntityValidationException;
+import eu.europeana.entitymanagement.definitions.model.Entity;
+import eu.europeana.entitymanagement.utils.EntityUtils;
+import eu.europeana.entitymanagement.vocabulary.EntityFieldsTypes;
+
+@Component
+public class EntityFieldsMinimalValidator implements ConstraintValidator<EntityFieldsMinimalValidatorInterface, Entity> {
+    
+    public void initialize(EntityFieldsMinimalValidatorInterface constraint) {
+//        System.out.println();
+    }
+
+    @Override
+	public boolean isValid(Entity entity, ConstraintValidatorContext context) {	
+    			
+		if (entity==null) {
+			return false;
+		}
+		
+		boolean returnValue=true;
+		
+		List<Field> entityFields = EntityUtils.getAllFields(entity.getClass());
+		
+		try {
+			for (Field field : entityFields) {
+				Object fieldValue = entity.getFieldValue(field);
+				if (fieldValue==null) { 
+					addConstraint(context, "The mandatory field: "+field.getName()+" cannot be NULL.");
```

Comment:
this contrain must be applied only for fields that are mandatory, based on the field cardinality 1..1 or 1..*

Reply:
Improved.